### PR TITLE
Added new slider to configure the current charge limit

### PR DESCRIPTION
**Pull request by fano0001**
I want to make it possible to set the current charge limit im Home Assistant for the NRGKick.

### CODE REVIEW

File: custom_components/nrgkick/number.py [hunk 1]
```diff
@@ -50,6 +53,18 @@ class NRGKickNumberEntityDescription(NumberEntityDescription, NRGKickRequiredKey
...
         api_fn=lambda c, v: c.set_charge_current_limit(v),
     ),
+    NRGKickNumberEntityDescription(
+        key="energy_limit_value",
+        translation_key="energy_limit_value",
+        device_class=NumberDeviceClass.CURRENT,
+        native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+        native_max_value= lambda data: data["cc_s"].energyLimit.max,
+        native_min_value= lambda data: data["cc_s"].energyLimit.min,
+        native_step=0.5,
```
Comment (dupondje) [inline custom_components/nrgkick/number.py:63]: In the App itself I can only change in steps of 1kWh, you tested that 0.5 is possible?

File: custom_components/nrgkick/websocket.py [hunk 2]
```diff
@@ -187,6 +188,27 @@ async def set_charge_current_limit(self, limit: float) -> bool:
...
+    async def set_energy_current_limit(self, limit: float) -> bool:
+        """Set the energy current limit."""
+        event = asyncio.Event()
+
+        request = self.__get_base_request()
+        request.header.type = nrgcp.Nrgcp.Header.Type.UPDATE
+        request.header.service = nrgcp.Nrgcp.Header.Service.CHARGE_CONTROL
+        request.header.property = nrgcp.Nrgcp.Header.Property.SETTINGS
+
+        request.payload.CHARGECONTROL_SETTINGS_UPDATE.energyLimit.limited = NrgcpTypes.EnergyLimitMode.ENERGY_LIMIT_MODE_LIMITED
```
Comment (dupondje) [inline custom_components/nrgkick/websocket.py:200]: It would be nice if we had a way to set it back to unlimited some way.

**Review by dupondje — CHANGES_REQUESTED** [review]
Thanks for this.
I've done a quick check, and have some smell remarks :)

File: custom_components/nrgkick/number.py (hunk 1 above)
Comment (fano0001) [reply custom_components/nrgkick/number.py:63]: No i did not tested it yet.
I will set the native_step to 1.0

File: custom_components/nrgkick/websocket.py (hunk 2 above)
Comment (fano0001) [reply custom_components/nrgkick/websocket.py:200]: Added a new switch to control ENERGY_LIMIT_MODE

**Comment by dupondje** [comment]
Had to do quite some fixes to get it working properly.
Please see https://github.com/dupondje/nrgkick-ha/pull/6/commits/6113e132805b9413385a17b78d4517d6b2797cec

It is always a good idea to have the code tested locally to see if everything works.

Thanks for your PR!

**Pull request merged**